### PR TITLE
Bragi: Add distance for features

### PR DESCRIPTION
**Pull request by benoit-bst**
Add distance between input coords and feature objects in json response, when we have request with {lat,lon} params like :

```
/autocomplete?q=14 juillet&_all_data=true&lat=48.526578&lon=2.679347
```
The corresponding response is :

```
...
          ],
          "feed_publishers": [
            {
              "id": "",
              "license": "",
              "name": "",
              "url": ""
            }
          ]
        }
      },
      "distance": 100                 <----------------------- in meters
    },
    {
      "type": "Feature",
      "geometry": {
        "coordinates": [
          2.56849,
          48.914572
...
```

### CODE REVIEW

File: libs/bragi/src/api.rs [hunk 1]
```diff
@@ -66,6 +70,17 @@ lazy_static! {
     ).unwrap();
 }
 
+fn add_distance(autocomp_resp: &mut model::Autocomplete, origin_coord: &Coord) {
+    for feature in &mut autocomp_resp.features {
+        if let ::geojson::Value::Point(p) = &feature.geometry.value {
+            if let [mut lon, mut lat] = p.as_slice() {
+                let feature_coord = Coord { lon: lon, lat: lat };
```
Comment (TeXitoi) [inline libs/bragi/src/api.rs]:
```rust
let feature_coord = Coord { lon, lat };
```

File: libs/bragi/src/api.rs [hunk 2]
```diff
@@ -362,7 +377,17 @@ impl ApiEndPoint {
...
+                    let mut response = model::v1::AutocompleteResponse::from(model_autocomplete);
+
+                    // Optional : add distance for each feature (in meters)
+                    if let Some(origin_coord) = &coord {
+                        if let model::v1::AutocompleteResponse::Autocomplete(autocomplete_resp) =
+                            &mut response
+                        {
+                            add_distance(autocomplete_resp, origin_coord);
+                        }
+                    }
```
Comment (TeXitoi) [inline libs/bragi/src/api.rs]:
```rust
use model::v1::AutocompleteResponse::Autocomplete;
if let (Some(coord), Autocomplete(autocomplete_resp)) = (&coord, &mut response) {
    add_distance(autocomplete_resp, origin_coord);
}
```

File: libs/bragi/src/lib.rs [hunk 3]
```diff
@@ -46,6 +46,8 @@ extern crate heck;
 extern crate urlencoded;
 extern crate valico;
 
+extern crate navitia_model;
```
Comment (TeXitoi) [inline libs/bragi/src/lib.rs]: Please, don't exclude this line from the others `extern crate`.

File: libs/bragi/src/model.rs [hunk 4]
```diff
@@ -347,7 +350,7 @@ pub struct Autocomplete {
     #[serde(rename = "type")]
     format_type: String,
     geocoding: Geocoding,
-    features: Vec<Feature>,
+    pub features: Vec<Feature>,
```
Comment (TeXitoi) [inline libs/bragi/src/model.rs]: Or everything is pub, or we add a getter.

File: tests/bragi_stops_test.rs [hunk 5]
```diff
@@ -345,3 +349,42 @@ fn stop_order_by_weight_test(bragi: &BragiHandler) {
...
+        let feature_first = features.first().unwrap();
+        assert!(feature_first.contains_key("distance"));
+        assert_eq!(feature_first["distance"], 100);
+
+        let feature_second = features.last().unwrap();
+        assert!(feature_second.contains_key("distance"));
+        assert_eq!(feature_second["distance"], 100);
+    }
+
+    // with out input coord
```
Comment (TeXitoi) [inline tests/bragi_stops_test.rs]: without in one word.

File: tests/bragi_stops_test.rs [hunk 6]
```diff
@@ -345,3 +349,42 @@ fn stop_order_by_weight_test(bragi: &BragiHandler) {
...
+        assert_eq!(features.len(), 2);
+
+        let feature_first = features.first().unwrap();
+        assert!(feature_first.contains_key("distance"));
+        assert_eq!(feature_first["distance"], 100);
+
+        let feature_second = features.last().unwrap();
+        assert!(feature_second.contains_key("distance"));
+        assert_eq!(feature_second["distance"], 100);
+    }
```
Comment (TeXitoi) [inline tests/bragi_stops_test.rs]: minor: you don't really need these blocks.

File: libs/bragi/src/api.rs (hunk 1 above)
Comment (benoit-bst) [reply libs/bragi/src/api.rs]: :white_check_mark:

File: libs/bragi/src/api.rs (hunk 2 above)
Comment (benoit-bst) [reply libs/bragi/src/api.rs]: :white_check_mark:

File: libs/bragi/src/lib.rs (hunk 3 above)
Comment (benoit-bst) [reply libs/bragi/src/lib.rs]: :white_check_mark:

File: tests/bragi_stops_test.rs (hunk 5 above)
Comment (benoit-bst) [reply tests/bragi_stops_test.rs]: :white_check_mark:

**Pull request merged**